File: generate_load_test_data.py
import json
import os
from datetime import datetime, timedelta

def generate_data():
    base_path = "data/stations/sensors_data/cardena"
    source_files = [
        os.path.join(base_path, "2026-02", "2026-02-04.json"),
        os.path.join(base_path, "2026-02", "2026-02-05.json")
    ]

    # Load source data
    data_samples = []
    for f in source_files:
        with open(f, 'r') as file:
            data_samples.append(json.load(file))

    start_date = datetime(2026, 1, 1)
    end_date = datetime(2027, 12, 31)
    
    current_date = start_date
    while current_date <= end_date:
        # Pick a sample (alternate based on day)
        sample_idx = current_date.day % len(data_samples)
        sample = data_samples[sample_idx]
        
        # Create a deep copy of the sample data to modify
        # Actually, we only need to modify fecha_raw
        new_data = json.loads(json.dumps(sample))
        
        date_str = current_date.strftime("%Y-%m-%d")
        month_str = current_date.strftime("%Y-%m")
        
        for element in new_data['data']:
            for sensor in element['sensores']:
                for entry in sensor['datos']:
                    # entry['fecha_raw'] is like "2026-02-04 23:57:00"
                    time_part = entry['fecha_raw'].split(' ')[1]
                    entry['fecha_raw'] = f"{date_str} {time_part}"
        
        # Save to file
        output_dir = os.path.join(base_path, month_str)
        os.makedirs(output_dir, exist_ok=True)
        output_file = os.path.join(output_dir, f"{date_str}.json")
        
        with open(output_file, 'w') as f:
            json.dump(new_data, f)
            
        current_date += timedelta(days=1)

if __name__ == "__main__":
    generate_data()

File: generate_load_test_data.R
library(tidyverse)
library(jsonlite)
library(lubridate)

generate_data <- function() {
  base_path <- "data/stations/sensors_data/cardena"
  source_files <- c(
    file.path(base_path, "2026-02", "2026-02-04.json"),
    file.path(base_path, "2026-02", "2026-02-05.json")
  )

  # Load source data
  data_samples <- map(source_files, \(f) read_json(f))

  start_date <- as_date("2026-01-01")
  end_date <- as_date("2027-12-31")

  all_dates <- seq(start_date, end_date, by = "day")

  walk(all_dates, \(current_date) {
    # Pick a sample (alternate based on day)
    # R is 1-indexed, so we use day_of_month mod len + 1
    sample_idx <- (day(current_date) %% length(data_samples)) + 1
    new_data <- data_samples[[sample_idx]]

    date_str <- format(current_date, "%Y-%m-%d")
    month_str <- format(current_date, "%Y-%m")

    # Update fecha_raw in the nested list structure
    new_data$data <- map(new_data$data, \(element) {
      element$sensores <- map(element$sensores, \(sensor) {
        sensor$datos <- map(sensor$datos, \(entry) {
          time_part <- str_split_i(entry$fecha_raw, " ", 2)
          entry$fecha_raw <- str_glue("{date_str} {time_part}")
          return(entry)
        })
        return(sensor)
      })
      return(element)
    })

    # Save to file
    output_dir <- file.path(base_path, month_str)
    if (!dir.exists(output_dir)) {
      dir.create(output_dir, recursive = TRUE)
    }

    output_file <- file.path(output_dir, str_glue("{date_str}.json"))
    write_json(
      new_data,
      output_file,
      auto_unbox = TRUE,
      pretty = TRUE
    )
  })
}

generate_data()
